plugins/module_utils/gandi_livedns_api.py

```diff
@@ -32,7 +32,8 @@ class GandiLiveDNSAPI(object):
 
     def __init__(self, module):
         self.module = module
-        self.api_key = module.params['api_key']
+        self.api_key = module.params.get('api_key')
```

Comment:
Why not keep the existing style?
```suggestion
        self.api_key = module.params['api_key']
```

Reply:
Right, I didn't know the key was there even when it wasn't specified in the task.